### ISSUE TITLE
Problem loading ImageNet labels after upgrading to 1.0.0-beta

### ISSUE THREAD

**Issue by schrum2** [deeplearning4j/deeplearning4j#5402]
This issue is sort of a continuation of #5266 but has to do with a completely different sub-section of the code. Specifically, the issue is with the Zoo models for Image Net. I think that there is a missing resource file that is supposed to contain all of the ImageNet class labels. I would assume that this would download automatically, but perhaps I just need to download it and store it in my project.

I think the resource I'm missing is: imagenet_class_index.json

The specific exception I'm getting is below. By the way ... why is ImageNetLabels.getLabels protected now? I had to do an annoying work-around that probably wasn't advisable, but I need that list of labels.

```
Could not load ImageNet labels
org.nd4j.shade.jackson.databind.JsonMappingException: No content to map due to end-of-input
 at [Source: UNKNOWN; line: 1, column: 1]
	at org.nd4j.shade.jackson.databind.JsonMappingException.from(JsonMappingException.java:148)
	at org.nd4j.shade.jackson.databind.ObjectMapper._initForReading(ObjectMapper.java:3607)
	at org.nd4j.shade.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:3547)
	at org.nd4j.shade.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:2648)
	at org.deeplearning4j.zoo.util.imagenet.ImageNetLabels.getLabels(ImageNetLabels.java:29)
	at org.deeplearning4j.zoo.util.BaseLabels.<init>(BaseLabels.java:22)
	at org.deeplearning4j.zoo.util.imagenet.ImageNetLabels.<init>(ImageNetLabels.java:22)
	at edu.southwestern.util.graphics.ImageNetClassification$AccessibleImageNetLabels.<init>(ImageNetClassification.java:40)
	at edu.southwestern.util.graphics.ImageNetClassification.getImageNetLabelsInstance(ImageNetClassification.java:55)
	at edu.southwestern.util.graphics.ImageNetClassification.bestLabel(ImageNetClassification.java:215)
	at org.deeplearning4j.examples.imagenet.ImageNetTest.resultsFromOne(ImageNetTest.java:164)
	at org.deeplearning4j.examples.imagenet.ImageNetTest.main(ImageNetTest.java:86)
```

**Comment by schrum2** [comment]
Also, sort of related, but where did the GoogLeNet model go? I had an old issue involving verification of this model ( https://github.com/deeplearning4j/deeplearning4j/issues/4139 ). Did the model get renamed, or did someone decide that it was in fact incorrect and needed to be removed?

**Comment by AlexDBlack** [comment]
I can run this in the examples repo (1.0.0-beta) without issue:
```
        ImageNetLabels iml = new ImageNetLabels();

        for( int i=0; i<5; i++ ){
            System.out.println(iml.getLabel(i));
        }
```
With output:
```
tench
goldfish
great_white_shark
tiger_shark
hammerhead
```

So I'm pretty sure it *is* present. How are you running your code - via an uber-jar? Via an IDE?
At any rate, since 1.0.0-beta it has been moved online and is downloaded on demand (instead of being distributed inside the JAR).

As for GoogLeNet - my understanding is that there were issues with the implementation, and it was deleted rather than have an incorrect version still around. (There was insufficient time, and it wasn't a high enough priority for us to fix it, just before release)

**Comment by schrum2** [comment]
The exception above is caused by running the following file from within an IDE:
https://github.com/schrum2/MM-NEAT/blob/dev/src/main/java/org/deeplearning4j/examples/imagenet/ImageNetTest.java

However, I have also run a different file via an uber-jar, and gotten essentially the same error via this class: https://github.com/schrum2/MM-NEAT/blob/dev/src/main/java/edu/southwestern/tasks/innovationengines/ImageNetBinMapping.java

I'm running my own code, not the one from your examples, but this code did work with a previous version of DL4J. I basically changed my pom file. However, unless the resource is downloaded automatically (as is done with the zoo models themselves in some cases), then I presumably just need to download it and put it in the right place in my code. I can't find imagenet_class_index.json though

**Comment by AlexDBlack** [comment]
@schrum2 Until very recently (and, including 1.0.0-beta) the actual resource was within the JAR itself.
It's definitely there, you can open the JAR and see for yourself:
![image](https://user-images.githubusercontent.com/2360237/40812441-3d336c96-6579-11e8-889c-fe998b4d4a2e.png)

This has already been changed on master to download to a local directory.

So, the only question here is what the difference is - why it works in tests and for me, but not for you.
Previously it used getResourceAsStream to load it - https://github.com/deeplearning4j/deeplearning4j/blob/883f23cb4f3bcbd23387e79b4aaa48313f2ec963/deeplearning4j/deeplearning4j-zoo/src/main/java/org/deeplearning4j/zoo/util/imagenet/ImageNetLabels.java#L29
Though maybe answering that is moot given it has already been changed...

Current version: https://github.com/deeplearning4j/deeplearning4j/blob/master/deeplearning4j/deeplearning4j-zoo/src/main/java/org/deeplearning4j/zoo/util/imagenet/ImageNetLabels.java#L33


Anyway, as for a workaround with 1.0.0-beta: you can try adding the imagenet_class_index.json file to your project resources, and see if it picks that one up.
Otherwise, double check that it is actually included in the uber-jar you are building.

**Comment by schrum2** [comment]
I solved the issue. The cause of the problem was that I was extending the ImageNetLabels class within a static inner class AccessibleImageNetLabels. I had the following inside of another file:

```
public static class AccessibleImageNetLabels extends ImageNetLabels {
	public AccessibleImageNetLabels() throws IOException {
		super();
	}
	public List<String> labels() throws IOException {
		return getLabels();
	}
}
```

When I moved the class into its own file (and removed "static") the image labels loaded just fine.

Of course, the only reason I made a hack/cheat class like this was that you made the getLabels() method protected in this release, so I extended the class to get access to the protected method. Regardless, this issue is fixed so I'll go ahead and close the issue.
